blockchain/state/statedb.go

```diff
@@ -397,12 +398,12 @@ func (s *StateDB) IsValidCodeFormat(addr common.Address) bool {
 	return false
 }
 
-// GetVmVersion return false when getStateObject(addr) or GetProgramAccount(stateObject.account) is failed.
+// GetVmVersion return false when getStateObject(addr) or GetProgramAccount(stateObject.account) fails and the account is EOA without code.
 func (s *StateDB) GetVmVersion(addr common.Address) (params.VmVersion, bool) {
 	stateObject := s.getStateObject(addr)
 	if stateObject != nil {
 		pa := account.GetProgramAccount(stateObject.account)
-		if pa != nil {
+		if pa != nil && !(bytes.Equal(pa.GetCodeHash(), emptyCodeHash) && pa.Type() == account.ExternallyOwnedAccountType) {
```

Comment:
```suggestion
		isEOA := pa.Type() == account.ExternallyOwnedAccountType
		isCodeEmpty = (pa != nil) && bytes.Equal(pa.GetCodeHash(), emptyCodeHash)
		if pa != nil && !(isEOA && isCodeEmpty) { // true for SCA or EOAWithCode, false for EOAWithoutCode
```

Reply:
I considered using an explanatory variable, but it would cause a panic if pa is nil. 
I hope that in the future, methods such as IsEOA will be added to the Account interface.

Reply:
Okay I'll put the explanatory method liike IsEOA in the backlog.